### ISSUE TITLE
Global CSS in modules

### ISSUE THREAD

**Issue by qfox** [gravity-ui/uikit#200]
Error occurred when trying to use uikit with Next.js.

Is there is a chance to support Next.js and reduce amount of peer deps some day?

---

How to reproduce:

1. Create new project with Next.js:
```
pnpm create next-app uikitapp -- --ts 
cd uikitapp
```

2. Install uikit:

<details>
<summary><code>pnpm install @yandex-cloud/uikit @yandex-cloud/i18n</code></summary>

```
Packages: +22
++++++++++++++++++++++
Progress: resolved 260, reused 248, downloaded 1, added 22, done

dependencies:
+ @yandex-cloud/i18n 0.6.0
+ @yandex-cloud/uikit 2.7.0

 WARN  Issues with peer dependencies found
.
└─┬ @yandex-cloud/uikit
  ├── ✕ unmet peer react@^16.0.0: found 18.1.0
  ├── ✕ unmet peer react-dom@^16.0.0: found 18.1.0
  ├── ✕ unmet peer @yandex-cloud/i18n@^0.4.0: found 0.6.0
  ├─┬ react-copy-to-clipboard
  │ └── ✕ unmet peer react@"^15.3.0 || ^16.0.0 || ^17.0.0": found 18.1.0
  ├─┬ react-popper
  │ └── ✕ unmet peer react@"^16.8.0 || ^17": found 18.1.0
  ├─┬ react-sortable-hoc
  │ ├── ✕ unmet peer react@"^16.3.0 || ^17.0.0": found 18.1.0
  │ └── ✕ unmet peer react-dom@"^16.3.0 || ^17.0.0": found 18.1.0
  ├─┬ react-virtualized-auto-sizer
  │ ├── ✕ unmet peer react@"^15.3.0 || ^16.0.0-alpha || ^17.0.0": found 18.1.0
  │ └── ✕ unmet peer react-dom@"^15.3.0 || ^16.0.0-alpha || ^17.0.0": found 18.1.0
  └─┬ react-window
    ├── ✕ unmet peer react@"^15.0.0 || ^16.0.0 || ^17.0.0": found 18.1.0
    └── ✕ unmet peer react-dom@"^15.0.0 || ^16.0.0 || ^17.0.0": found 18.1.0
```

</details>

React 18 issue: #178

3. Add page, run server

```
cat > pages/uikit.tsx << BODY
import type { NextPage } from 'next'
import {Button} from '@yandex-cloud/uikit';

const Page: NextPage = () => {
  return (
    <div>
      <Button view="action" size="l">Hey, you!</Button>
    </div>
  )
}

export default Page
BODY
npm run dev
```

4. Curl a page

<details>
<summary><code>curl http://localhost:3000/uikit | xpath -e '/html/body/*/text()' | jq -r '.err.stack'</code></summary>

```
Found 1 nodes in stdin:
-- NODE --
ModuleError: Module Error (from ./node_modules/.pnpm/next@12.1.6_react-dom@18.1.0+react@18.1.0/node_modules/next/dist/build/webpack/loaders/error-loader.js):
Global CSS cannot be imported from within node_modules.
Read more: https://nextjs.org/docs/messages/css-npm
Location: node_modules/.pnpm/@yandex-cloud+uikit@2.7.0_2d73508a1935c0d6c393a21d5bb53228/node_modules/@yandex-cloud/uikit/build/esm/components/ArrowToggle/ArrowToggle.js
    at Object.emitError (/Users/qfox/free/uikit/node_modules/.pnpm/next@12.1.6_react-dom@18.1.0+react@18.1.0/node_modules/next/dist/compiled/webpack/bundle5.js:49443:6)
    at Object.ErrorLoader (/Users/qfox/free/uikit/node_modules/.pnpm/next@12.1.6_react-dom@18.1.0+react@18.1.0/node_modules/next/dist/build/webpack/loaders/error-loader.js:24:10)
    at LOADER_EXECUTION (/Users/qfox/free/uikit/node_modules/.pnpm/next@12.1.6_react-dom@18.1.0+react@18.1.0/node_modules/next/dist/compiled/webpack/bundle5.js:140726:14)
    at runSyncOrAsync (/Users/qfox/free/uikit/node_modules/.pnpm/next@12.1.6_react-dom@18.1.0+react@18.1.0/node_modules/next/dist/compiled/webpack/bundle5.js:140727:4)
    at iterateNormalLoaders (/Users/qfox/free/uikit/node_modules/.pnpm/next@12.1.6_react-dom@18.1.0+react@18.1.0/node_modules/next/dist/compiled/webpack/bundle5.js:140844:2)
    at /Users/qfox/free/uikit/node_modules/.pnpm/next@12.1.6_react-dom@18.1.0+react@18.1.0/node_modules/next/dist/compiled/webpack/bundle5.js:140817:4
    at /Users/qfox/free/uikit/node_modules/.pnpm/next@12.1.6_react-dom@18.1.0+react@18.1.0/node_modules/next/dist/compiled/webpack/bundle5.js:49669:15
    at eval (eval at create (/Users/qfox/free/uikit/node_modules/.pnpm/next@12.1.6_react-dom@18.1.0+react@18.1.0/node_modules/next/dist/compiled/webpack/bundle5.js:141841:10), <anonymous>:14:1)
    at /Users/qfox/free/uikit/node_modules/.pnpm/next@12.1.6_react-dom@18.1.0+react@18.1.0/node_modules/next/dist/build/webpack/plugins/profiling-plugin.js:116:29
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

</details>

**Comment by amje** [comment]
UIKit is heavily depend on our internal build pipeline, which includes css bundling in that way. We are looking forward to change it, switch to css-in-js or to all-in-one css file.

About reducing peer deps. In the next major i18n library will be carried inside.

**Comment by qfox** [comment]
Feels like any ideas and suggestions will be appreciated. right?

**Comment by amje** [comment]
Yes, sure

**Comment by resure** [comment]
It should be possible to use UIKit with next.js with a help of `next-transpile-modules` package:

```javascript
// next.config.js
const withTM = require('next-transpile-modules')(['@yandex-cloud/uikit']);

const nextConfig = {
    reactStrictMode: true,
};

module.exports = withTM(nextConfig);
```

**Comment by ogonkov** [comment]
@qfox react 18 typings was merged in #224

**Comment by amje** [comment]
Closing this. Seems all tasks are done and uikit is working in Next.js now
